sysdeps/managarm/generic/file.cpp

```diff
@@ -1262,8 +1267,14 @@ int sys_select(int num_fds, fd_set *read_set, fd_set *write_set,
 	}
 
 	struct epoll_event evnts[16];
-	int n = epoll_wait(fd, evnts, 16,
-			timeout ? (timeout->tv_sec * 1000 + timeout->tv_usec / 10) : -1);
+	int n;
+	if(sigmask != NULL) {
+		n = epoll_pwait(fd, evnts, 16,
+			timeout ? (timeout->tv_sec * 1000 + timeout->tv_nsec / 100) : -1, sigmask);
+	} else {
+		n = epoll_wait(fd, evnts, 16,
+			timeout ? (timeout->tv_sec * 1000 + timeout->tv_nsec / 100) : -1);
+	}
```

Comment:
Pointless code duplication? Both branches can call into `epoll_pwait` and pass `sigmask` directly (`epoll_pwait` with a NULL signal mask is equivalent to `epoll_wait`)

Reply:
Good point, I forgot about that. I’ll change that